Feature/#162 Make CoatRack (database + config) deployable by using a build script or a docker-compose command (#37)
BrunoSilvaATB/coatrack: README.md

![CoatRack architecture overview](./spring-boot/admin/src/main/resources/static/images/coatrack-architecture-overview.png)

> ## Installation for local usage

Reviewer: "local usage" seems misleading, because the goal of this approach is that it can be used anywhere. 

So I would just change back to "Install". 

The previous text under "install" was more like a placeholder, as it was not about installation at all. That stuff should be moved to another page or another subsection.

Author: Done. I changes the title to 'Install' again and moved the development part to the CoatRack wiki.

Go to the docker-compose-setup folder and execute:

 ```sh

>  .init.sh

Reviewer: I assume this should be:
```suggestion
 ./init.sh
```

Author: Done.

 ```

> The script only needs to be executed once at the very first time to initialize the docker volumes. After that, executing 'docker-compose up/down' is sufficient. After starting up, the CoatRack web application will be accessible at `http://localhost:8080`.

Reviewer: If it is only required once, I would separate into two commands: 
```
./init.sh
docker-compose up
```
I would put that 'docker-compose up/down' also as commands to execute, e.g.
"[...]
After the initialisation, use the following command to start CoatRack:
```
docker-compose up
```
"

Author: Done as suggested.

> If you want to reset CoatRack, stop/remove all containers, remove all volumes and start the ```init.sh``` script again.

Reviewer: Just an idea: does it maybe make sense to create a "un-install" script for this?

Author: I think it is a nice and easy to implement feature which is why I just added it.

> ## Installation for development

Reviewer: This is actually not an "installation", it was just a placeholder. So I would either move this into the "Usage" section or to the "developers" page in the wiki.

Author: Done.

> At first you have to build the project:

Reviewer: I liked the old text ;)

```suggestion
Scripts for building and running CoatRack are available in the root folder of the code base. To build CoatRack, use:
```

./run-admin-application.sh
The script only needs to be executed once at the very first time to initialize the docker volumes. After that, executing 'docker-compose up/down' is sufficient. After starting up, the CoatRack web application will be accessible at `http://localhost:8080`.

If you want to reset CoatRack, stop/remove all containers, remove all volumes and start the ```init.sh``` script again.

## Installation for development

At first you have to build the project:

```sh
.build.sh 
```

> After starting up, the CoatRack web application will be accessible at `http://localhost:8080` 

Reviewer: This sentence should not get lost ;)

Author: It is kept in the current Wiki page.

> When you want to run CoatRack you have to execute the script below. You also have the option to use a custom Spring profile, which can be passed to the script as follows:

Reviewer: ```suggestion
To run your locally built CoatRack admin web application, use the following command (you also have the option to use a custom Spring profile by passing its name to the start script):
```

## Install

> Go to the docker-compose-setup folder and execute:

Reviewer: Before this I would suggest to add a short "pre-conditions" list for the installation, e.g. a short bullet point list including java 11 or later, maven 3.? or later, docker and docker compose...

Also it would be good to add a short explanation of what the init script does (maybe also as bullet points): build coatrack, create a local docker image, create volumes, etc.

 ```sh

>  sh init.sh

Reviewer: "init" sounds to me like it only sets some local variables, path config etc.

Maybe a more descriptive name should be used, e.g. `build-and-init-docker-volumes.sh` or `init-local-docker-compose-setup.sh` or `install-dockerized-coatrack.sh`?

Reviewer: or maybe simply `install.sh`?
